Mathlib/NumberTheory/ModularForms/Basic.lean

```diff
@@ -230,7 +230,7 @@ def mul {k_1 k_2 : ℤ} {Γ : Subgroup SL(2, ℤ)} (f : ModularForm Γ k_1) (g :
     -- Porting note: was `by simpa using ...`
     -- `mul_slash_SL2` is no longer a `simp` and `simpa only [mul_slash_SL2] using ...` failed
     rw [SlashInvariantForm.coe_mul, mul_slash_SL2]
-    exact (f.bdd_at_infty' A).mul (g.bdd_at_infty' A)
+    exact Filter.BoundedAtFilter.mul (f.bdd_at_infty' A) (g.bdd_at_infty' A)
```

Comment:
I looked at that and thought, "why change the dot-notation back to the clumsier older notation"? On checking out the code and playing with it, I realised that the deprecated alias `IsBoundedAtImInfty.mul` is "masking" the more general lemma `BoundedAtFilter.mul` so it doesn't get found by namespace search. This, to me, seems like a good reason _not_ to add a deprecation for the old lemma. Then this proof can become simply
```lean
  bdd_at_infty' A := by simpa only [coe_mul, mul_slash_SL2] 
    using (f.bdd_at_infty' A).mul (g.bdd_at_infty' A)```

Reply:
I am unable to get your code working even after un-deprecating the alias, so I tried an alternative, see if it's okay.

Reply:
I tried it out before posting it, so it definitely does work.

Reply:
I just tried it again, and it works fine (after removing the section `BasicOperations` in the other file which you have subsequently added). 

Perhaps it was not clear from my old message that I meant "remove `IsBoundedAtImInfty.mul` and don't deprecate it", not "leave the lemma there and don't deprecate it".